src/wasm-interpreter.h

```diff
@@ -1919,6 +1919,33 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
     }
     return Literal(int32_t(data->values.size()));
   }
+  Flow visitStringConcat(StringConcat* curr) { 
+    NOTE_ENTER("StringConcat");
+    Flow flow = visit(curr->left);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto left = flow.getSingleValue();
+    flow = visit(curr->right);
+    if (flow.breaking()) {
+      return flow;
+    }
+    auto right = flow.getSingleValue();
+    NOTE_EVAL2(left, right);
+    auto leftData = left.getGCData();
```

Comment:
There should be a null check before these, see other places with `trap("null ref");` in this file.